### ISSUE TITLE
Optimize SetVkContext

### ISSUE THREAD

**Issue by Michael-Lfx** [Think-Silicon/GLOVE#50]
https://github.com/Think-Silicon/GLOVE/blob/52bf1f88d180920e0269d1e6a81655817ea10008/GLES/source/vulkan/context.cpp#L54

Thinking about the following things, **is there any possible to remove `SetVkContext()` from all the resources model class and just call `vulkanAPI::GetContext()` inside `source/vulkan` layer without passing it to `resources/` layer?**
1. There is just one global `vkContext_t` here
1. Almost all the mobile devices currently don't have multiple GPUs
2. Even if there are multiple GPUs, OpenGL ES 2.0 ~ 3.2 does not provide interface to choose different GPU to interact with.

**Comment by abasilak** [comment]
Yes it is an issue that struggled us a lot in the past, since we setting this value in every newly created object. However it is not first in our priority list right now.
